build-system/tasks/coverage-map/index.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const argv = require('minimist')(process.argv.slice(2));
+const fs = require('fs').promises;
+const log = require('fancy-log');
+const {dist} = require('../dist');
+const {installPackages} = require('../../common/utils');
+const {startServer, stopServer} = require('../serve');
+
+let puppeteer;
+let explore;
+
+const coverageJsonName = argv.json || 'out.json';
+const testUrl =
+  argv.url || 'http://localhost:8000/examples/everything.amp.html';
+const outHtml = argv.html || 'out.html';
+
+async function collectCoverage() {
+  log(`Opening browser and navigating to ${testUrl}...`);
+  const browser = await puppeteer.launch({
+    defaultViewport: {width: 1200, height: 800},
+  });
+  const page = await browser.newPage();
+
+  // Enable JavaScript coverage
+  await page.coverage.startJSCoverage();
+  // Navigate to page
+  await page.goto(testUrl);
+  await page.waitFor(() => !!document.querySelector('style[amp-runtime]'));
+  await page.waitFor(
+    () =>
+      !!document.body &&
+      getComputedStyle(document.body).visibility === 'visible'
```

Comment:
Is there a reason this `waitFor` needs to be split in two instead of chaining the conditionals themselves? I'm imagining something like this:
```javascript
await page.waitFor(
  () => 
    !!document.querySelector('style[amp-runtime]') &&
    !!document.body &&
    getComputedStyle(document.body).visibility === 'visible'
);
```

Reply:
That's true. I guess I had them separated just to make clear of what Dima described earlier. I'll combine them.